src/clothing_mod.h

```diff
@@ -45,14 +49,84 @@ struct clothing_mod {
     float get_mod_val( const clothing_mod_type &type, const item &it ) const;
     bool has_mod_type( const clothing_mod_type &type ) const;
 
+    /**
+     * Check whether the mod applies a given flag.
+     *
+     * @param f a flag string to check
+     * @return bool true if the mod would apply the flag
+     */
+    bool applies_flag( const std::string &f ) const;
+
+    /**
+     * Check whether the mod suppresses a given flag.
+     *
+     * @param f a flag string to check
+     * @return bool true if the mod would suppress the flag
+     */
+    bool suppresses_flag( const std::string &f ) const;
+
+    /**
+     * Whether the mod applies any flags when used.
+     *
+     * @return bool true if the mod applies flags
+     */
+    bool applies_flags( ) const;
+
+    /**
+     * Whether the mod suppresses any flags when used.
+     *
+     * @return bool true if the mod applies flags
+     */
+    bool suppresses_flags( ) const;
+
+    /**
+     * Checks if the mod is compatible with an item.
+     *
+     * Flags are incompatible if an item has one of the flags this mod applies,
+     * or if this mod excludes one of its flags.
+         *
+         * Body parts are compatible if the item covers at least one of the valid_parts
+         * and does not cover any of the invalid_parts.
+         *
```

Comment:
Looks like the indent got messed up here somehow
```suggestion
     *
     * Body parts are compatible if the item covers at least one of the valid_parts
     * and does not cover any of the invalid_parts.
     *
```